provider/property_override.rb

```diff
@@ -0,0 +1,82 @@
+# Copyright 2017 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'api/object'
+require 'api/type'
+
+module Provider
+  # Override a resource property (Api::Type) in api.yaml
+  class PropertyOverride < Api::Object
```

Comment:
This class is virtually identical to `Provider::ResourceOverride`. Let's refactor this can make both derive from a base class and share all the code that was copied over.

Reply:
It is actually slightly different, for instance, `ensure_property_fields` calls send(:include, overriden) on `Api::Type` instead of `Api::Resource`. Same in `our_overrides_module` and `require_module`. We could modify these method so that the type to call is given by a method we have to override (same way we did with overridden).

There is also the naming that I changed a bit calling the properties 'fields' instead. I did that because having a method called `ensure_property_properties` seems poorly named. We could rename it to Fields in the Api::Resource but I didn't want to cause to much disruption because you were actively working on Chef/Puppet.

I suggest we merge this PR without this change and I do the merge in a separate PR. This will allow me to work on releasing new resources while we get this right. Since it won't change how we define terraform.yaml, I think it is a reasonable approach. I added a TODO.

Reply:
The term "property" has nothing to do with the API properties, but with Ruby's definition of a property (and when you have attr_* functions). You may call them attributes too :)

I think merging them into one is a good thing. We can rename the methods accordingly. Let's deal with them in the next PR then.